doc/shared/sundials/Errors.rst

```diff
@@ -0,0 +1,186 @@
+  :c:func:`SUNContext_GetLastError` to check these functions for errors.
+
+
+.. _SUNDIALS.Errors.Handlers:
+
+Error Handler Functions
+-----------------------
+
+Errors that occur internally to SUNDIALS result in an error handler function being called. 
+These error handler functions have the type
+
+.. c:type:: typedef int (*SUNErrHandlerFn)(int line, const char* func, const char* file,
+                                           const char* msg, SUNErrCode err_code,
+                                           void* err_user_data, SUNContext sunctx);
+
+SUNDIALS provides a few different error handlers that can be used, or a custom one defined by the
+user can be provided (useful for linking SUNDIALS errors to your application's error handling).
+The default error handler is :c:func:`SUNLogErrHandlerFn` which logs an error to a specified
+file or ``stderr`` if no file is specified.
+
+The error handlers provided in SUNDIALS are:
+
+.. c:function:: void SUNLogErrHandlerFn(int line, const char* func, const char* file, \
+                                        const char* msg, SUNErrCode err_code, \
+                                        void* err_user_data, SUNContext sunctx)
+
+  Logs the error that occurred using the :c:type:`SUNLogger` for ``sunctx``.
+  This is the default error handler.
+
+  :param line: the line number at which the error occured
+  :param func: the function in which the error occured
+  :param file: the file in which the error occured 
+  :param msg: the message to log, if this is ``NULL`` then the default error message for the error code will be used
+  :param err_code: the error code for the error that occured
+  :param err_user_data: the user pointer provided to :c:func:`SUNContext_PushErrHandler`
+  :param sunctx: pointer to a valid :c:type:`SUNContext` object
+
+  :return: ``void``
+
+.. c:function:: void SUNAbortErrHandlerFn(int line, const char* func, const char* file, \
+                                          const char* msg, SUNErrCode err_code, \
+                                          void* err_user_data, SUNContext sunctx)
+
+  Logs the error and aborts the program if an error occured.
+
+  :param line: the line number at which the error occured
+  :param func: the function in which the error occured
+  :param file: the file in which the error occured 
+  :param msg: the message to log, if this is ``NULL`` then the default error message for the error code will be used
+  :param err_code: the error code for the error that occured
+  :param err_user_data: the user pointer provided to :c:func:`SUNContext_PushErrHandler`
+  :param sunctx: pointer to a valid :c:type:`SUNContext` object
+
+  :return: ``void``
+
+
+.. c:function:: void SUNAssertErrHandlerFn(int line, const char* func, const char* file, \
+                                           const char* stmt, SUNErrCode err_code, \
+                                           void* err_user_data, SUNContext sunctx)
```

Comment:
No one cannot be removed, because the distinct message is important, particularly to the assert version since it shows the statement that was asserted.

Reply:
I didn't see that a different error handler is called for asserts vs other errors. Should the difference in the message being passed to the handler be in `SUNAssert` or `SUNCheck`?

Reply:
Ah, you are right. Since we changed `SUNAssert` to call `SUNCheck` if it is enabled otherwise state an assumption, these Assert handlers are no longer needed.

Reply:
I removed them